exchanges/poloniex/poloniex_wrapper.go

```diff
@@ -417,172 +522,365 @@ func (e *Exchange) GetWithdrawalsHistory(ctx context.Context, c currency.Code, _
-	if err := action.Validate(); err != nil {
-		return nil, err
+	if action == nil {
+		return nil, common.ErrNilPointer
 	}
-
-	oID, err := strconv.ParseInt(action.OrderID, 10, 64)
-	if err != nil {
+	if err := action.Validate(); err != nil {
 		return nil, err
 	}
-
-	resp, err := e.MoveOrder(ctx,
-		oID,
-		action.Price,
-		action.Amount,
-		action.TimeInForce.Is(order.PostOnly),
-		action.TimeInForce.Is(order.ImmediateOrCancel))
-	if err != nil {
-		return nil, err
+	if action.AssetType != asset.Spot {
+		return nil, fmt.Errorf("%w: asset type: %v", asset.ErrNotSupported, action.AssetType)
 	}
-
-	modResp, err := action.DeriveModifyResponse()
+	tif, err := TimeInForceString(action.TimeInForce)
 	if err != nil {
 		return nil, err
 	}
-	modResp.OrderID = strconv.FormatInt(resp.OrderNumber, 10)
-	return modResp, nil
-}
-
-// CancelOrder cancels an order by its corresponding ID number
-func (e *Exchange) CancelOrder(ctx context.Context, o *order.Cancel) error {
-	if err := o.Validate(o.StandardCancel()); err != nil {
-		return err
-	}
-
-	orderIDInt, err := strconv.ParseInt(o.OrderID, 10, 64)
-	if err != nil {
-		return err
-	}
-
-	return e.CancelExistingOrder(ctx, orderIDInt)
-}
-
-// CancelBatchOrders cancels an orders by their corresponding ID numbers
+	switch action.Type {
+	case order.Market, order.Limit, order.LimitMaker:
+		resp, err := e.CancelReplaceOrder(ctx, &CancelReplaceOrderRequest{
+			orderID:       action.OrderID,
+			ClientOrderID: action.ClientOrderID,
+			Price:         action.Price,
+			Quantity:      action.Amount,
+			AmendedType:   action.Type.String(),
+			TimeInForce:   tif,
+		})
+		if err != nil {
+			return nil, err
+		}
+		modResp, err := action.DeriveModifyResponse()
+		if err != nil {
+			return nil, err
+		}
+		modResp.OrderID = resp.ID
+		return modResp, nil
+	case order.Stop, order.StopLimit:
+		oTypeString, err := OrderTypeString(action.Type)
+		if err != nil {
+			return nil, err
+		}
+		oResp, err := e.CancelReplaceSmartOrder(ctx, &CancelReplaceSmartOrderRequest{
+			orderID:          action.OrderID,
+			ClientOrderID:    action.ClientOrderID,
+			Price:            action.Price,
+			StopPrice:        action.TriggerPrice,
+			Amount:           action.Amount,
+			AmendedType:      oTypeString,
+			ProceedOnFailure: !action.TimeInForce.Is(order.ImmediateOrCancel),
+			TimeInForce:      tif,
+		})
+		if err != nil {
+			return nil, err
+		}
+		modResp, err := action.DeriveModifyResponse()
+		if err != nil {
+			return nil, err
+		}
+		modResp.OrderID = oResp.ID
+		return modResp, nil
+	default:
+		return nil, fmt.Errorf("%w %v", order.ErrUnsupportedOrderType, action.Type)
+	}
+}
+
+// CancelOrder cancels an order by its corresponding ID number
+func (e *Exchange) CancelOrder(ctx context.Context, o *order.Cancel) error {
+	if o.OrderID == "" && o.ClientOrderID == "" {
+		return order.ErrOrderIDNotSet
+	}
+	if err := o.Validate(o.StandardCancel()); err != nil {
```

Comment:
```suggestion
	if err := o.Validate(); err != nil {
```

As you have done the check more appropriately above